src/app/components/MeasurementExplorer.tsx

```diff
@@ -23,20 +25,40 @@ export function MeasurementExplorer(props: MeasurementExplorerProps) {
     kind = '1d',
   } = props;
   function defaultInfo(dataIndex: number) {
+    const varNames = new Set(Object.keys(data[dataIndex].variables));
     return {
       dataIndex,
-      xVariableName: Object.keys(data[dataIndex].variables).includes('x')
-        ? 'x'
-        : Object.keys(data[dataIndex].variables)[0],
-      yVariableName: Object.keys(data[dataIndex].variables).includes('y')
-        ? 'y'
-        : Object.keys(data[dataIndex].variables)[1],
+      xVariableName: varNames.has('x') ? 'x' : varNames[0],
+      yVariableName: varNames.has('y') ? 'y' : varNames[1],
     };
   }
   const [info, setInfo] = useState<ExplorerInfo>({
     flipHorizontalAxis: false,
     ...defaultInfo(0),
   });
+  /* variables for this measurement are mapped into `id - label (units)`
+    `id` necessary bc files may have repeated labels
+  */
+  function dropdownVariables(axis: 'x' | 'y') {
+    function formatVar(varKey: string) {
+      const { label, units } = variables[varKey];
+      const formatUnit = units ? ` (${units})` : '';
+      const formatVarKey = `${varKey} - `;
+      return formatVarKey + label + formatUnit;
+    }
+    const { variables } = data[info.dataIndex];
+    const oppositeAxis: XOrY = axis === 'x' ? 'yVariableName' : 'xVariableName';
```

Comment:
You can remove `XOrY` type
`oppositeAxis` is already defined as `'yVariableName' | 'xVariableName'`